Flaky-test: PartitionedProducerConsumerTest. testPartitionedTopicInterceptor

PartitionedProducerConsumerTest. testPartitionedTopicInterceptor is flaky. It fails sporadically.

[example failure](https://github.com/apache/pulsar/runs/7432986412?check_suite_focus=true)

```
  Error:  Tests run: 33, Failures: 7, Errors: 0, Skipped: 15, Time elapsed: 35.84 s <<< FAILURE! - in org.apache.pulsar.client.api.PartitionedProducerConsumerTest
  Error:  testPartitionedTopicInterceptor(org.apache.pulsar.client.api.PartitionedProducerConsumerTest)  Time elapsed: 1.263 s  <<< FAILURE!
  org.apache.pulsar.client.api.PulsarClientException: java.util.concurrent.ExecutionException: org.apache.pulsar.client.api.PulsarClientException: The client is not connected to the broker when unsubscribing the subscription my-partitioned-subscriber of the topic persistent://my-property/my-ns/interceptor-partitionedtopic1-1658332461875-partition-7
  	at org.apache.pulsar.client.api.PulsarClientException.unwrap(PulsarClientException.java:1074)
  	at org.apache.pulsar.client.impl.ConsumerBase.unsubscribe(ConsumerBase.java:682)
  	at org.apache.pulsar.client.api.PartitionedProducerConsumerTest.testPartitionedTopicInterceptor(PartitionedProducerConsumerTest.java:1029)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
  Caused by: java.util.concurrent.ExecutionException: org.apache.pulsar.client.api.PulsarClientException: The client is not connected to the broker when unsubscribing the subscription my-partitioned-subscriber of the topic persistent://my-property/my-ns/interceptor-partitionedtopic1-1658332461875-partition-7
  	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
  	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
  	at org.apache.pulsar.client.impl.ConsumerBase.unsubscribe(ConsumerBase.java:677)
  	... 13 more
  Caused by: org.apache.pulsar.client.api.PulsarClientException: The client is not connected to the broker when unsubscribing the subscription my-partitioned-subscriber of the topic persistent://my-property/my-ns/interceptor-partitionedtopic1-1658332461875-partition-7
  	at org.apache.pulsar.client.impl.ConsumerImpl.unsubscribeAsync(ConsumerImpl.java:422)
  	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
  	at java.base/java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3612)
  	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
  	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
  	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
  	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
  	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
  	at org.apache.pulsar.client.impl.MultiTopicsConsumerImpl.unsubscribeAsync(MultiTopicsConsumerImpl.java:602)
  	... 14 more
```

--- a/pulsar-broker/src/test/java/org/apache/pulsar/client/api/PartitionedProducerConsumerTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/client/api/PartitionedProducerConsumerTest.java
@@ -1025,6 +1025,8 @@ public class PartitionedProducerConsumerTest extends ProducerConsumerBase {
         assertEquals(newProducerPartitions.get(), newPartitions);
         assertEquals(newConsumerPartitions.get(), newPartitions);
 
+        Awaitility.await().untilAsserted(() -> consumer.isConnected());
+
         producer.close();
         consumer.unsubscribe();
         consumer.close();

